test/token/ERC20WithPermit.test.js

```diff
@@ -0,0 +1,1102 @@
+                    recipient.address,
+                    amount
+                  )
+              ).to.be.revertedWith("Transfer amount exceeds balance")
+            })
+          })
+        })
+
+        context(
+          "when the spender does not have enough approved balance",
+          () => {
+            const allowance = initialSupply.sub(1)
+
+            beforeEach(async () => {
+              await token
+                .connect(initialHolder)
+                .approve(anotherAccount.address, allowance)
+            })
+
+            context("when the token owner has enough balance", () => {
+              const amount = initialSupply
+
+              it("should revert", async () => {
+                await expect(
+                  token
+                    .connect(anotherAccount)
+                    .transferFrom(
+                      initialHolder.address,
+                      recipient.address,
+                      amount
+                    )
+                ).to.be.revertedWith("Transfer amount exceeds allowance")
+              })
+            })
+
+            context("when the token owner does not have enough balance", () => {
+              const amount = initialSupply
+
+              beforeEach(async () => {
+                await token
+                  .connect(initialHolder)
+                  .transfer(anotherAccount.address, 1)
+              })
+
+              it("should revert", async () => {
+                await expect(
+                  token
+                    .connect(anotherAccount)
+                    .transferFrom(
+                      initialHolder.address,
+                      recipient.address,
+                      amount
+                    )
+                ).to.be.revertedWith("Transfer amount exceeds allowance")
+              })
+            })
+
+            context("when the token owner is the zero address", () => {
+              const allowance = initialSupply
+
+              it("should revert", async () => {
+                await expect(
+                  token
+                    .connect(anotherAccount)
+                    .transferFrom(ZERO_ADDRESS, recipient.address, allowance)
+                ).to.be.revertedWith("Transfer amount exceeds allowance")
+              })
+            })
+          }
+        )
+      })
+
+      context("when the recipient is the zero address", () => {
+        const allowance = initialSupply
+
+        beforeEach(async () => {
+          await token
+            .connect(initialHolder)
+            .approve(anotherAccount.address, allowance)
+        })
+
+        it("should revert", async () => {
+          await expect(
+            token
+              .connect(anotherAccount)
+              .transferFrom(initialHolder.address, ZERO_ADDRESS, allowance)
+          ).to.be.revertedWith("Transfer to the zero address")
+        })
+      })
+
+      context("when given the maximum allowance", () => {
```

Comment:
This should be moved under `context("when the recipient is not the zero address", () => {`.